internal/store/engine_state.go

```diff
@@ -41,10 +43,18 @@ type EngineState struct {
 
 	PermanentError error
 
+	// The user has indicated they want to exit
+	Exit bool
+
+	// The full log stream for tilt. This might deserve gc or file storage at some point.
+	Log *bufsync.ThreadSafeBuffer
```

Comment:
shouldn't we just Rlock the state and copy this when we want to read from it? we don't want individual pieces of enginestate to have their own mutexes; it will get confusing and deadlock-prone fast